### PR TITLE
feat: Initialize

### PR DESCRIPTION
**Pull request by karelianpie**
Closes https://github.com/Major-Eth/yearn-web-lib/issues/1

Init the repo to be a NPM package.

* React with Typescript;
* Storybook
* Styled components

Added a sample `Button` just to make sure storybook is working fine, it's not yet the https://github.com/Major-Eth/yearn-web-lib/issues/2

Please let me know if you want to add/remove any dependencies or configs

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -0,0 +1,15 @@
+# dependencies
+node_modules
+
+# production
+dist
+
+# misc
+.DS_Store
+
```
Comment (Major-Eth) [inline .gitignore:9]: Could you add a `.env` and a `env.list`, please ? 🙏

File: package.json [hunk 2]
```diff
@@ -0,0 +1,52 @@
+{
+  "name": "@yfi/yearn-web-lib",
```
Comment (Major-Eth) [inline package.json:2]: After some talk, it will be `@yearn/yearn-web-lib` 😄

File: package.json [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+  "version": "0.1.0",
+  "description": "Basic web library for Yearn's projects",
+  "license": "MIT",
```
Comment (Major-Eth) [inline package.json:5]: Let's remove that for now 🙏

File: tsconfig.json [hunk 4]
```diff
@@ -0,0 +1,74 @@
+{
+  "compilerOptions": {
+    /* Visit https://aka.ms/tsconfig.json to read more about this file */
+
+    /* Basic Options */
+    // "incremental": true,                         /* Enable incremental compilation */
+    "target": "es5",                                /* Specify ECMAScript target version: 'ES3' (default), 'ES5', 'ES2015', 'ES2016', 'ES2017', 'ES2018', 'ES2019', 'ES2020', 'ES2021', or 'ESNEXT'. */
```
Comment (Major-Eth) [inline tsconfig.json:7]: Could we target `ES2020` or `ESNEXT` ?

File: tsconfig.json [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+    "module": "commonjs",                           /* Specify module code generation: 'none', 'commonjs', 'amd', 'system', 'umd', 'es2015', 'es2020', or 'ESNext'. */
```
Comment (Major-Eth) [inline tsconfig.json:8]: Could we use `es2020`, or `ESNext` ?

**Review by Major-Eth — COMMENTED** [review]
Hey !
Great, some small comments just to sync, but good start 🚀

File: package.json (hunk 2 above)
Comment (karelianpie) [reply package.json]: Changed with https://github.com/Major-Eth/yearn-web-lib/pull/8/commits/fabe51e1ad7d1dcf65d0cf2a8b581dda07f73c59

File: .gitignore (hunk 1 above)
Comment (karelianpie) [reply .gitignore:9]: Definitely, done with https://github.com/Major-Eth/yearn-web-lib/pull/8/commits/bbf7452976845a1d860eb13220782111b5cedf7b

File: package.json (hunk 3 above)
Comment (karelianpie) [reply package.json]: The license? I actually didn't add it initially, but then there was this warning about the license missing that was annoying, happy to remove. Removed with https://github.com/Major-Eth/yearn-web-lib/pull/8/commits/4cad7725edb01f0cd3a7d9020e8f444ad4d02cc4

```cmd
➜  yearn-web-lib git:(issue-1) ✗ yarn
yarn install v1.22.17
warning package.json: No license field
warning @yearn/yearn-web-lib@0.1.0: No license field
```

File: tsconfig.json (hunk 4 above)
Comment (karelianpie) [reply tsconfig.json]: Set it to `ESNEXT` with https://github.com/Major-Eth/yearn-web-lib/pull/8/commits/dba3032a8f09d5d48d3759d79ff43511368db906

File: tsconfig.json (hunk 5 above)
Comment (karelianpie) [reply tsconfig.json]: Set it to `ESNext ` with https://github.com/Major-Eth/yearn-web-lib/pull/8/commits/dba3032a8f09d5d48d3759d79ff43511368db906

Had to also set the module resolution to `node`

**Review by Major-Eth — APPROVED** [review]
LGTM, let's proceed with this !

**Pull request merged**